signer/migrations/0004__create_functions.sql

```diff
@@ -44,3 +44,57 @@ BEGIN
     FROM blockchain as blocks;
 END;
 $$ LANGUAGE plpgsql;
+
+-- Table-Valued Function (TVF) for fetching a Stacks blockchain from a given
+-- stacks block hash and bitcoin block hash, only looking back the `max_depth`
+-- number of bitcoin blocks. The bitcoin blocks are used to consider only
+-- stacks blocks anchored to the canonical bitcoin chain.
+--
+-- - stacks_chain_tip: The stacks block hash to start the blockchain from ("chain tip").
+-- - bitcoin_chain_tip: The bitcoin block hash to start the canonical blockchain from ("chain tip").
+-- - max_depth: The maximum depth of the blockchain to fetch.
+CREATE FUNCTION sbtc_signer.stacks_blockchain_of (
+    stacks_chain_tip BYTEA,
+    bitcoin_chain_tip BYTEA,
+    max_depth INT
+)
+RETURNS TABLE (
+    block_hash BYTEA,
+    parent_hash BYTEA,
+    block_height BIGINT
+) 
+AS $$
+BEGIN
+    RETURN QUERY
+    WITH RECURSIVE bc_blocks AS (
+        SELECT * FROM sbtc_signer.bitcoin_blockchain_of(bitcoin_chain_tip, max_depth)
+    ),
+    blockchain AS (
+        SELECT
+            stacks_blocks.block_hash,
+            stacks_blocks.block_height,
+            stacks_blocks.parent_hash
```

Comment:
style nit: can we use leading commas please.

Reply:
Done in [c3e0400](https://github.com/stacks-network/sbtc/pull/747/commits/c3e04003d7fb9578c919756d7ed254180c3df45a), but we should settle on some tool+config to consistent format SQL (that is not only available in a specific IDE, ideally 😅 )